btk/catalog.py

```diff
@@ -70,33 +70,62 @@ def from_file(cls, catalog_file, verbose=False):
     def _prepare_table(self):
         table = deepcopy(self._raw_catalog)
 
+
         # TODO: does the WLDCatalog require the 'ra' and 'dec' columns
         # convert ra dec from degrees to arcsec in catalog.
-        if "ra" in table.columns:
-            table["ra"] *= 3600
-        if "dec" in table.columns:
+        if "ra" in table:
+            table["Ra"] *= 3600
+        if "dec" in table:
             table["dec"] *= 3600
 
+        f = self._raw_catalog["fluxnorm_bulge"] / (
+                self._raw_catalog["fluxnorm_disk"] + self._raw_catalog["fluxnorm_bulge"]
+        )
+        r_sec = np.hypot(
+            self._raw_catalog["a_d"] * (1 - f) ** 0.5 * 4.66, self._raw_catalog["a_b"] * f ** 0.5 * 1.46
+        )
+        # BTK now requires ref_mags, but WLD still wants magnitudes
+        table["ref_mag"] = self._raw_catalog["i_ab"]
+        table["btk_size"] = r_sec
+        #ADds the extra columns to both catalogs just to be sure
+        self._raw_catalog["ref_mag"] = self._raw_catalog["i_ab"]
+        self._raw_catalog["btk_size"] = r_sec
+
         return table
 
 
 class CosmosCatalog(Catalog):
     @classmethod
-    def from_file(cls, catalog_file, verbose=False):
+    def from_file(cls, catalog_files, verbose=False):
+        """
+
+        Paramters
+        ---------
+        catalog_files: list of galsim cataolgs
+        """
         # This will return a COSMOSCatalog object.
-        catalog = galsim.COSMOSCatalog(file_name=catalog_file)
+        catalog_coord = astropy.table.Table.read(catalog_files[0], 'fits')
+        catalog_fit = astropy.table.Table.read(catalog_files[1], 'fits')
+        catalog = astropy.table.hstack([catalog_coord, catalog_fit])
 
         return cls(catalog, verbose=verbose)
 
     def _prepare_table(self):
-        table = astropy.table.Table(self._raw_catalog.real_cat)
-
+        table = deepcopy(self._raw_catalog)
         # make notation for 'ra' and 'dec' standard across code.
         table.rename_column("RA", "ra")
         table.rename_column("DEC", "dec")
+        table.rename_column("MAG", "ref_mag")
+        index = np.where(t["IDENT_1"] == self.raw_catalog["IDENT_1"] for t in table)
 
         # convert ra dec from degrees to arcsec in catalog.
         table["ra"] *= 3600
         table["dec"] *= 3600
 
+        size = self._raw_catalog["flux_radius"]*self._raw_catalog["PIXEL_SCALE"]
+        table["btk_size"] = size
+        table["btk_index"] = index
+        #ADds the extra columns to both catalogs just to be sure
+        self._raw_catalog["ref_mag"] = self._raw_catalog["MAG"]
+        self._raw_catalog["btk_size"] = size
```

Comment:
Since you are using many columns in the table, there's a high chance of failure if one of them is missing from the input cat.
I guess we should have a list of mandatory columns, check if they're all present and fail with a nice exception message if not. Especially if you're using case sensitive column names.

Reply:
Well that's the problem with handling the catalog. You never know what columns you're gonna get and you therefore need a custom `Catalog` class for each catalog.

Reply:
That's true, but if users want to use a new catalog, they would need to do this work anyway ; at least for the supported catalog it saves them the trouble from creating the table themselves (and they will ultimately get the results in a standard way)